### ISSUE TITLE
[Bug]: 

### ISSUE THREAD

**Issue by craigvc** [beyondcode/herd-community#429]
### Operating system version

Mac OS Sonoma 14.3.1 (M3)

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.4.1

### PHP Version

8.2.16

### Bug description

I am trying to debug from phpstorm
I have herd pro.
I check my php-v and get this
➜  Herd php -v
PHP 8.2.16 (cli) (built: Feb 16 2024 00:30:02) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.16, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.16, Copyright (c), by Zend Technologies
    
    so it seems xdebug isnt installed?
    how do i fix this?
    

### Steps to reproduce

Install PHP storm, install herd on MAC Ventura

**Comment by mpociot** [comment]
The CLI version will only activate Xdebug when you run "herd debug". 
You can use this as a "php" command replacement, like "herd debug -v"

**Comment by craigvc** [comment]
ah so i need the browser extension for debugging a site? I thought the pro version did it automatically?

**Comment by mpociot** [comment]
The pro version automatically enables Xdebug on-the-fly, either by setting a breakpoint in your PHPStorm project, or if you don't use PHPStorm, by toggling the Xdebug extension.

Be sure to save the file after setting a breakpoint, so that Herd can detect it.

**Comment by craigvc** [comment]
Maybe I am missing something and apologies if its dumb questions!
Ok so i set a breakpoint on phpstrom, listen for connections in phpstorm
Then visit herd site - it should just work yes?
Do i need to install a browser extension or something?

**Comment by mpociot** [comment]
Yeah, that's how it should work. You should be able to verify this by returning `phpinfo()`.
With a breakpoint, xdebug should be enabled.

One thing you might want to look out for: We detect your breakpoint by looking for a folder in your parked paths, that matches the hostname.
If you linked your site with a different name, this detection might not work and you need to use the extension instead.

**Comment by craigvc** [comment]
The paths are not changed from the site name btw

**Comment by mpociot** [comment]
Just to double check, if you use the Xdebug browser extension, it works as expected?

**Comment by craigvc** [comment]
no doesnt work there either?

**Comment by mpociot** [comment]
Weird. Could you please toggle the checkbox in the Debugger settings? Maybe that setting was not properly persisted

**Comment by craigvc** [comment]
ah getting somewhere!
It may be caused by path mappings misconfiguration or not synchronized local and remote projects.  To figure out the problem check path mappings configuration for 'gcc-v2.test' server at PHP|Servers or enable 'Break at first line in PHP scripts' option (from Run menu). Do not show again

**Comment by craigvc** [comment]
<img width="1882" alt="image" src="https://github.com/beyondcode/herd-community/assets/13494843/c8e339fe-8f02-47f0-88f3-6449f4dfb768">

**Comment by craigvc** [comment]
Any clues to this error?
I moved the project now to the Herd folder instead of my normal folder and its now throwing this error so its getting something from herd!

(with and without the chrome extension)

**Comment by marvinosswald** [comment]
> Weird. Could you please toggle the checkbox in the Debugger settings? Maybe that setting was not properly persisted

this fixed it for me 👍

**Comment by craigvc** [comment]
> > Weird. Could you please toggle the checkbox in the Debugger settings? Maybe that setting was not properly persisted
> 
> this fixed it for me 👍

I tried this several times

**Comment by mpociot** [comment]
@craigvc what happens when you enable "break at first line" in PHPStorm, as the error message suggests?

**Comment by craigvc** [comment]
it brings me to this page in php storm dump-loader.php
<?php

use BeyondCode\HerdConfiguration\HerdConfiguration;

/*
|--------------------------------------------------------------------------
| Laravel Herd
|--------------------------------------------------------------------------
|
| This script is the entrypoint of your Laravel Herd request. If you see
| this file while using Herd's Xdebug auto-detection, you may want to
| modify your PHPStorm settings, to not automatically break at the
| first line.
|
| Learn more about it in the Herd documentation
| https://herd.laravel.com/docs/1/herd-pro/xdebug
|
*/

require_once __DIR__ . '/HerdConfiguration.php';

try {
    $config = HerdConfiguration::load();
} catch (\Throwable $e) {
    return;
}

**Comment by philipmclifton** [comment]
I am also having this issue again after originally finding success with the break at first breakpoint. dumping phpinfo shows that Xdebug is not loaded in the request at all even when breakpoints exist.  and checking the workspace.yml file to confirm the following exists:

```
 <component name="XDebuggerManager">
    <breakpoint-manager>
      <breakpoints>
        <line-breakpoint enabled="true" type="php">
          <url>file://$PROJECT_DIR$/app/Traits/GiftCards/Orders/InteractsWithTemporaryOrderRows.php</url>
          <line>11</line>
          <option name="timeStamp" value="1" />
        </line-breakpoint>
        <line-breakpoint enabled="true" type="php">
          <url>file://$PROJECT_DIR$/app/Traits/GiftCards/Orders/InteractsWithTemporaryOrderRows.php</url>
          <line>9</line>
          <option name="timeStamp" value="2" />
        </line-breakpoint>
        <line-breakpoint enabled="true" type="php">
          <url>file://$PROJECT_DIR$/app/Traits/GiftCards/Orders/InteractsWithTemporaryOrderRows.php</url>
          <line>13</line>
          <option name="timeStamp" value="3" />
        </line-breakpoint>
      </breakpoints>
    </breakpoint-manager>
  </component>
```

**Comment by philipmclifton** [comment]
Ok - Toggling the checkbox in the herd debug settings page worked for me this time!

**Comment by craigvc** [comment]
@mpociot I am using filamentphp and livewire could that have anything do with it?
I see like 122 notifications about the Debug session was finished without being paused when i reload a page in the browser (I see it in PHPStorm)

**Comment by paulmichaeldev** [comment]
> it brings me to this page in php storm dump-loader.php
> 
> ```
> 
> use BeyondCode\HerdConfiguration\HerdConfiguration;
> 
> /*
> |--------------------------------------------------------------------------
> | Laravel Herd
> |--------------------------------------------------------------------------
> |
> | This script is the entrypoint of your Laravel Herd request. If you see
> | this file while using Herd's Xdebug auto-detection, you may want to
> | modify your PHPStorm settings, to not automatically break at the
> | first line.
> |
> | Learn more about it in the Herd documentation
> | https://herd.laravel.com/docs/1/herd-pro/xdebug
> |
> */
> 
> require_once __DIR__ . '/HerdConfiguration.php';
> 
> try {
>     $config = HerdConfiguration::load();
> } catch (\Throwable $e) {
>     return;
> }
> ```

I'm also experiencing this - I am able to step through it to my actual breakpoint, but I get the feeling there's an option in PHPStorm that is a little wonky.

Every request served by PHP gets halted at line 20 in /Applications/Herd.app/Contents/Resources/valet/dump-loader.php.

My project is Vue with Inertia, so I _think_ that might rule out filament/livewire @craigvc

**Comment by ju5t** [comment]
I'm using Livewire but unfortunately xdebug is not working for me either. It keeps breaking at the dump-loader script. Break at first line in script is disabled.

**Comment by craigvc** [comment]
@mpociot can we get a little guidance here ?

**Comment by craigvc** [comment]
@paulmichaeldev I did read that can actually be normal in the sense its sending lots of ajax requests - so would apply from inertia and livewire perspective. Issue i am stuck at is i cant even step through or it doesnt detect my breakpoints - phpstorm EAP and 2023.3.4

**Comment by mpociot** [comment]
@craigvc if it stops for you in the dump-loader.php file, this is the default behavior of PHPStorm as it breaks on the first line.
You should be able to just continue debugging / running the script using the PHPStorm debugger. At this point, xdebug is already activated
